src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala

```diff
@@ -0,0 +1,66 @@
+package io.treeverse.clients
+
+import io.treeverse.committed.Committed.RangeData
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.io.Text
+import org.apache.hadoop.mapred._
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat
+
+import java.io.File
+
+class EntryRecordReader(ssTableReader: SSTableReader, localFile: String) extends RecordReader[Text, Text] {
+  private val it = ssTableReader.getData(localFile)
+
+  override def next(key: Text, value: Text): Boolean = {
+    if (!it.hasNext) {
+      return false
+    }
+    key.set(it.next().key)
+    true
+  }
+
+  override def createKey(): Text = {
+    new Text()
+  }
+
+  override def createValue(): Text = {
+    new Text()
+  }
+
+  override def getPos: Long = {
+    0 // TODO(johnnyaug) complete
+  }
+
+  override def close(): Unit = {
+    new File(localFile).delete
+  }
+
+  override def getProgress: Float = {
+    0 // TODO(johnnyaug) complete
+  }
+}
+
+object LakeFSInputFormat {}
+
+class LakeFSInputFormat extends InputFormat[Text, Text] {
+  override def getSplits(job: JobConf, numSplits: Int): Array[InputSplit] = {
+    val metaRangePath = job.get(FileInputFormat.INPUT_DIR)
+    val p = new Path(metaRangePath)
+    val fs = p.getFileSystem(job)
+    val localFile = File.createTempFile("lakefs", "metarange")
```

Comment:
Add a TODO to configure (just) a lakeFS server endpoint, and ask that server about filenames.